### PR TITLE
Add classes for array and slice literals

### PR DESCRIPTION
**Pull request by owen-mc**

### CODE REVIEW

File: ql/src/semmle/go/Expr.qll [hunk 1]
```diff
@@ -437,6 +437,60 @@ class StructLit extends CompositeLit {
...
+ * [6]int{1, 2, 3, 5}
+ * [...]string{"Sat", "Sun"}
+ * []int{1, 2, 3, 5}
+ * []string{"Sat", "Sun"}
+ * ```
+ */
+class ArrayOrSliceLit extends CompositeLit {
+  Type type;
+
+  ArrayOrSliceLit() { getType() instanceof ArrayType or getType() instanceof SliceType }
```
Comment (max-schaefer) [inline ql/src/semmle/go/Expr.qll:456]: Did you perhaps mean to bind the field `type` here?

```suggestion
  ArrayOrSliceLit() { type = getType() and (type instanceof ArrayType or type instanceof SliceType) }
```

File: ql/src/semmle/go/Expr.qll [hunk 2]
```diff
@@ -437,6 +437,60 @@ class StructLit extends CompositeLit {
...
+ * Examples:
+ *
+ * ```go
+ * [10]string{}
+ * [6]int{1, 2, 3, 5}
+ * [...]string{"Sat", "Sun"}
+ * ```
+ */
+class ArrayLit extends ArrayOrSliceLit {
+  ArrayLit() { getType() instanceof ArrayType }
```
Comment (max-schaefer) [inline ql/src/semmle/go/Expr.qll:473]: If you bind `type` as suggested above, you can now do

```suggestion
  override ArrayType type;
```

File: ql/src/semmle/go/Expr.qll [hunk 3]
```diff
@@ -437,6 +437,60 @@ class StructLit extends CompositeLit {
...
+ *
+ * Examples:
+ *
+ * ```go
+ * []int{1, 2, 3, 5}
+ * []string{"Sat", "Sun"}
+ * ```
+ */
+class SliceLit extends CompositeLit {
+  SliceLit() { getType() instanceof SliceType }
```
Comment (max-schaefer) [inline ql/src/semmle/go/Expr.qll:489]:
```suggestion
  override SliceType type;
```

File: ql/src/semmle/go/Expr.qll (hunk 1 above)
Comment (max-schaefer) [reply ql/src/semmle/go/Expr.qll:456]: Of course you can also just remove the field altogether and leave the characteristic predicates as they are.

**Comment by owen-mc** [comment]
I've added a commit to update the strings for map and struct literals as well

**Pull request merged**